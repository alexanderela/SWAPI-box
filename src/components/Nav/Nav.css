/*.button-container {
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex-direction: column;
  padding-right: 16px;
  top: 80px;
  width: 80%;
}*/

.Nav {
  align-items: center;
  bottom: 0px;
  display: flex;
  height: 5rem;
  justify-content: space-around;
  padding-right: 16px;
  top: 80px;
  transform: translatex(0%);
  transition: transform 0.2s ease;
  width: 80%;
}

.nav {
  text-decoration: none;
  color: white;
}

.nav:hover {
  color: #000;
}

.cat-button {
  background: #000;
  border: 1px solid #FFF;
  border-radius: 0px 100px 100px 0px;
  color: #fff;
  cursor: pointer;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 2em;
  font-weight: 600;
  height: 3rem;
  margin: 16px 0px;
  padding: 0px;
  padding-left: 0px;
  transition: 
    width 0.5s ease,
    padding 0.2s ease, 
    color 0.2s ease, 
    text-shadow 0.2s ease;
  width: 12rem;
}

.cat-button-active {
  background-color: #363535;
  box-shadow: #d9ffd9 0px 0px 15px 2px;
}

.cat-button-inactive {
  background-color: #000;
}

.cat-button:hover {
  border-left: gray double 15px;
  color:black;
  cursor: pointer;
  padding:0px 16px 0px 0px;
}

.cat-button:focus {
  outline: 0;
}

.people-button:hover {
  background: #d9ffd9;
  box-shadow: #d9ffd9 0px 0px 20px 2px;
  text-shadow: 
    #0F0 0px 0px 6px, 
    #0F0 0px 0px 10px;
}

.planets-button:hover {
  background:  #ffd9d9;
  box-shadow: #ffd9d9 0px 0px 20px 2px;
  text-shadow: 
    #F00 0px 0px 6px, 
    #F00 0px 0px 10px;
}

.vehicles-button:hover {
  background: #ffffd9;
  box-shadow: #ffffd9 0px 0px 20px 2px;
  text-shadow: 
    #FF0 0px 0px 6px, 
    #FF0 0px 0px 10px;
}
