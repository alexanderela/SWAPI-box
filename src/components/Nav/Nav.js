import React, {Component} from 'react';
import './Nav.css'
import FavoriteButton from '../FavoriteButton/FavoriteButton.js';
import PropTypes from 'prop-types';


class Nav extends Component {
	constructor(props) {
		super(props)
		this.state = {
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
		}
	}

	handlePeopleClick = (e) => {
		if (this.state.peopleSelected === true) {
			const { name } = e.target
			this.setState({ peopleSelected: false })
			this.props.toggleCategoryState(name)
		} else {
			this.setState({ 
				peopleSelected: true,      
				vehiclesSelected: false,
      	planetsSelected: false 
			})	
			this.props.showPeople()
		}
	}

	handlePlanetClick = (e) => {
		const { name } = e.target
		if (this.state.planetsSelected === true) {
			this.setState({ planetsSelected: false })
			this.props.toggleCategoryState(name)
		} else {
			this.setState({ 
				peopleSelected: false,      
				vehiclesSelected: false,
				planetsSelected: true 
			})	
			this.props.showPlanet()
		}
	}

	handleVehicleClick = (e) => {
		const { name } = e.target
		if (this.state.vehiclesSelected === true) {
			this.setState({ vehiclesSelected: false })
			this.props.toggleCategoryState(name)
		} else {
			this.setState({ 				
				peopleSelected: false,      
				vehiclesSelected: true,
				planetsSelected: false, 
			})	
			this.props.showVehicle()
		}
	}

render() {
	const { showPeople, showVehicles, showPlanets } = this.props
	const { peopleSelected, planetsSelected, vehiclesSelected } = this.state
	
	return(
		<div className="button-container">
		    <button 
		      className={`cat-button people-button 
		      	${peopleSelected 
		      		? "cat-button-active" 
		      		: "cat-button-inactive" }`}
		      name="people"
		      onClick={this.handlePeopleClick}
		    >People</button>
		    <button 
		      className={`cat-button planets-button 
		      	${planetsSelected 
		      		? "cat-button-active" 
		      		: "cat-button-inactive" }`}
		      name="planets"
		      onClick={this.handlePlanetClick}
		    >Planets</button>
		    <button 
		      className={`cat-button vehicles-button 
		      	${vehiclesSelected 
		      		? "cat-button-active" 
		      		: "cat-button-inactive" }`}
		      name="vehicles"
		      onClick={this.handleVehicleClick}
		    >Vehicles</button>
		    <FavoriteButton className="FavoriteButton" />
		  </div>
		)
	}
}

Nav.propTypes = {
	showPeople: PropTypes.func,
	showVehicle: PropTypes.func,
	showPlanet: PropTypes.func
}

export default Nav;