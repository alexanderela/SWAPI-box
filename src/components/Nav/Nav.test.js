import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import { shallow, mount } from 'enzyme';
import people from '../../mockData/mockPeople.js';

describe('Nav', () => {
	let wrapper;
	let mockFunction;

	beforeEach(() => {
    mockFunction = jest.fn()
		wrapper = shallow(<Nav   
						showPeople={mockFunction}
            showPlanet={mockFunction}
            showVehicle={mockFunction}/>)
	})

	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('activates People category on click of People button', () => {
		const expectedState = {
      peopleSelected: true,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.find('.people-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

	it('deactivates People category on 2nd click of People button', () => {
		const expectedState = {
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.setState({ peopleSelected: true })
		wrapper.find('.people-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

	it('activates Planets category on click of Planets button', () => {
		const expectedState = {
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: true
		}
		wrapper.find('.planets-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

	it('deactivates Planets category on 2nd click of Planets button', () => {
		const expectedState = {
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.setState({ planetsSelected: true })
		wrapper.find('.planets-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

	it('activates Vehicles category on click of Vehicles button', () => {
		const expectedState = {
      peopleSelected: false,
      vehiclesSelected: true,
      planetsSelected: false
		}
		wrapper.find('.vehicles-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

	it('deactivates Vehicles category on 2nd click of Vehicles button', () => {
		const expectedState = {
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.setState({ vehiclesSelected: true })
		wrapper.find('.vehicles-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})
	// it('invokes handleVehicleClick on click of Vehicles button', () => {
	// 	wrapper .find.simulate click
	// 	expect mockfunction to be called
	// })
})