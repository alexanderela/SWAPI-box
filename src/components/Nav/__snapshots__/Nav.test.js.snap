// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches the snapshot 1`] = `undefined`;
